rest_server.go

```diff
@@ -154,29 +156,32 @@ func (a *restServer) installRoutes(routesInfo map[int][]RouteInfo) {
+	a.multiplexer.Put(a.cfg.restServer.apiPrefix+"/:category/:id", a.makeHandler(handleUpdate))
+	a.multiplexer.Patch(a.cfg.restServer.apiPrefix+"/:category/:id", a.makeHandler(handlePatch))
+	a.multiplexer.Delete(a.cfg.restServer.apiPrefix+"/:category/:id", a.makeHandler(handleDelete))
+	a.multiplexer.Get(a.cfg.restServer.apiPrefix+"/:category", a.makeHandler(handleRetrieveMany))
+	a.multiplexer.Get(a.cfg.restServer.apiPrefix+"/:parentcategory/:id/:category", a.makeHandler(handleRetrieveMany))
+	a.multiplexer.Post(a.cfg.restServer.apiPrefix+"/:category", a.makeHandler(handleCreate))
+	a.multiplexer.Post(a.cfg.restServer.apiPrefix+"/:parentcategory/:id/:category", a.makeHandler(handleCreate))
+	a.multiplexer.Head(a.cfg.restServer.apiPrefix+"/:category", a.makeHandler(handleInfo))
+	a.multiplexer.Head(a.cfg.restServer.apiPrefix+"/:parentcategory/:id/:category", a.makeHandler(handleInfo))
 
 	// versioned routes
-	a.multiplexer.Get("/v/:version/:category/:id", a.makeHandler(handleRetrieve))
-	a.multiplexer.Put("/v/:version/:category/:id", a.makeHandler(handleUpdate))
-	a.multiplexer.Patch("/v/:version/:category/:id", a.makeHandler(handlePatch))
-	a.multiplexer.Delete("/v/:version/:category/:id", a.makeHandler(handleDelete))
-	a.multiplexer.Get("/v/:version/:category", a.makeHandler(handleRetrieveMany))
-	a.multiplexer.Get("/v/:version/:parentcategory/:id/:category", a.makeHandler(handleRetrieveMany))
-	a.multiplexer.Post("/v/:version/:category", a.makeHandler(handleCreate))
-	a.multiplexer.Post("/v/:version/:parentcategory/:id/:category", a.makeHandler(handleCreate))
-	a.multiplexer.Head("/v/:version/:category", a.makeHandler(handleInfo))
-	a.multiplexer.Head("/v/:version/:parentcategory/:id/:category", a.makeHandler(handleInfo))
-
+	a.multiplexer.Get(a.cfg.restServer.apiPrefix+"/v/:version/:category/:id", a.makeHandler(handleRetrieve))
+	a.multiplexer.Put(a.cfg.restServer.apiPrefix+"/v/:version/:category/:id", a.makeHandler(handleUpdate))
+	a.multiplexer.Patch(a.cfg.restServer.apiPrefix+"/v/:version/:category/:id", a.makeHandler(handlePatch))
+	a.multiplexer.Delete(a.cfg.restServer.apiPrefix+"/v/:version/:category/:id", a.makeHandler(handleDelete))
+	a.multiplexer.Get(a.cfg.restServer.apiPrefix+"/v/:version/:category", a.makeHandler(handleRetrieveMany))
+	a.multiplexer.Get(a.cfg.restServer.apiPrefix+"/v/:version/:parentcategory/:id/:category", a.makeHandler(handleRetrieveMany))
+	a.multiplexer.Post(a.cfg.restServer.apiPrefix+"/v/:version/:category", a.makeHandler(handleCreate))
+	a.multiplexer.Post(a.cfg.restServer.apiPrefix+"/v/:version/:parentcategory/:id/:category", a.makeHandler(handleCreate))
+	a.multiplexer.Head(a.cfg.restServer.apiPrefix+"/v/:version/:category", a.makeHandler(handleInfo))
+	a.multiplexer.Head(a.cfg.restServer.apiPrefix+"/v/:version/:parentcategory/:id/:category", a.makeHandler(handleInfo))
+
+	for customRoute, f := range a.customHandlers() {
+		a.multiplexer.Handle(a.cfg.restServer.customRoutePrefix+customRoute, f)
```

Comment:
Same here this is too fragile. Use a join to make sure you always start with a / and there is always another / separating them.

Reply:
replaced with path.Join everywhere